### DISCUSSION TITLE
Touch or click

### DISCUSSION THREAD

**Discussion by charlenni** [Mapsui/Mapsui#2369] [Development]
We use sometimes click, sometimes touch as wording for the same. Perhaps we should use one and that everywhere?

Any comments?

**Comment by janusw** [comment]
> We use sometimes click, sometimes touch as wording for the same. Perhaps we should use one and that everywhere?

I'd vote for using more touchscreen-oriented language, like MAUI and other mobile-first frameworks.

But I think the analogy to a "click" would be a "tap". In my understanding "touch" is more general and includes other touch gestures like pan, pinch, swipe, etc.

See https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/gestures/tap (and neighboring pages).

However, note that also MAUI does not seem to be 100% consistent about this: A button is still "clicked", not "tapped".
https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.button.clicked

**Comment by charlenni** [comment]
You are right, click and tap are the same. In this case, I normally would prefer click to tap, because I'm too old.

But lets see, what the others think.

**Comment by pauldendulk** [comment]
I asked chatgpt about a generic term for finger and mouse and it came up with Pointer https://chat.openai.com/share/fa86b218-8957-4ee8-9614-1fd9ff4ba484. Since this term is also used in Blazor (which I worked with at that time) that is what I used in this PR in the develop/4.1 branch: https://github.com/Mapsui/Mapsui/pull/2326/files. 

So, PointerUp might be an option although a PointerUp event is not always a click/tap event. So maybe we need a different name for that.

My preference is to follow the main trend, but I don't know what the main trend is 😁. Blazor uses it, which is a big one, and relatively new but I don't know about other frameworks. I have seen tap used as well. Perhaps we should look into some other frameworks like Avalonia and Uno, or Angular and React.
